test/tests/amqp/test_amqp_node_rediscover.py

```diff
@@ -9,138 +9,222 @@
 from time import sleep
-import Queue
+import gevent
+import gevent.queue
 import random
 import socket
 import flogging
-import logging
-import pika
 import unittest
 import json
-import threading
+import ipaddress
+import netifaces
 from BaseHTTPServer import HTTPServer, BaseHTTPRequestHandler
 import fit_common
 import test_api_utils
 from nose.plugins.attrib import attr
+from nosedep import depends
+
 logs = flogging.get_loggers()
-amqp_queue = Queue.Queue(maxsize=0)
-webhook_port = 9889
-nodefound_id = ""
-webhook_received = False
-webhook_body = ""
-
-
-class AmqpWorker(threading.Thread):
-    '''
-    This AMQP worker Class will creat another thread when initialized and runs asynchronously.
-    The external_callback is the callback function entrance for user.
-    The callback function will be call when AMQP message is received.
-    Each test case can define its own callback and pass the function name to the AMQP class.
-    The timeout parameter specify how long the AMQP daemon will run. self.panic is called when timeout.
-    eg:
-    def callback(self, ch, method, properties, body):
-        logs.debug(" [x] %r:%r" % (method.routing_key, body))
-        logs.debug(" [x] %r:%r" % (method.routing_key, body))
-
-    td = fit_amqp.AMQP_worker("node.added.#", callback)
-    td.setDaemon(True)
-    td.start()
-
-    TODO:
-    The common AMQP related test module of FIT is being refactoring. This AMQP test class is only for temporary use.
-    It will be obsolete and replaced by a common AMQP test module.
-    '''
-
-    def __init__(self, exchange_name, topic_routing_key, external_callback, timeout=10):
-        threading.Thread.__init__(self)
-        pika_logger = logging.getLogger('pika')
-        if fit_common.VERBOSITY >= 8:
-            pika_logger.setLevel(logging.DEBUG)
-        elif fit_common.VERBOSITY >= 4:
-            pika_logger.setLevel(logging.WARNING)
-        else:
-            pika_logger.setLevel(logging.ERROR)
-        amqp_port = fit_common.fitports()['amqp_ssl']
-        self.connection = pika.BlockingConnection(
-            pika.ConnectionParameters(host=fit_common.fitargs()["rackhd_host"], port=amqp_port))
-        self.channel = self.connection.channel()
-        result = self.channel.queue_declare(exclusive=True)
-        queue_name = result.method.queue
-        self.channel.queue_bind(exchange=exchange_name, queue=queue_name, routing_key=topic_routing_key)
-        self.channel.basic_consume(external_callback, queue=queue_name)
-        self.connection.add_timeout(timeout, self.dispose)
+WEBHOOK_PORT = 9889
 
-    def dispose(self):
-        logs.debug_7('Pika connection timeout')
-        if self.connection.is_closed is False:
-            self.channel.stop_consuming()
-            self.connection.close()
-        self.thread_stop = True
 
-    def run(self):
-        logs.debug_7('start consuming')
-        self.channel.start_consuming()
+class _TstHostInterfacer(object):

```

Comment:
Let's open another story to do that. It makes total sense, but making this routine "public" is really a different thing than this PR.